src/NHapi.Base/Util/SynchronizedCache.cs

```diff
@@ -0,0 +1,144 @@
+﻿#if NETFRAMEWORK
+
+namespace NHapi.Base.Util
+{
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Threading;
+
+    internal class SynchronizedCache<TKey, TValue> : IEnumerable<KeyValuePair<TKey, TValue>>
+    {
+        private readonly ReaderWriterLockSlim readerWriterLock = new ReaderWriterLockSlim();
+        private readonly Dictionary<TKey, TValue> dictionary = new Dictionary<TKey, TValue>();
+
+        ~SynchronizedCache()
+        {
+            if (readerWriterLock != null)
+            {
+                readerWriterLock.Dispose();
+            }
```

Comment:
Can we simplify this to `readerWriterLock?.Dispose()`?

Reply:
It can, I was going to do that change anyway but must have forgotten ... I will make the change.